<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">temp</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p><strong>Last updated:</strong> 2017-08-14</p>
<p><strong>Code version:</strong> 3e56f29e38fdb990fe1da0920d5c0f6f6e54a8de</p>
<div id="sweet-potato-project" class="section level1">
<h1>Sweet Potato Project</h1>
<p>Much of this code doesn’t work anymore because of modifications to the <a href="https://github.com/dcgerard/updog"><code>updog</code></a> package</p>
<ul>
<li><a href="snp_analysis.html">First analysis of SNP data</a></li>
<li><a href="iter_procedure.html">Look at iterative procedure and effect of uniform prior on parents</a>. This results in a non-uniform prior on the children, as can be expected.</li>
<li><a href="updot_overdispersion.html">First try at overdispersed updog</a></li>
<li><a href="more_updog_over.html">Run overdispersed updog on all of the datasets Felipe gave me</a></li>
<li><a href="updog_overdispersion_seq_est.html">Look at estimating sequencing error rate and results look promising. No more errors when I bound the overdispersion parameter.</a></li>
<li><a href="more_updog_over_seq_est.html">Run through all datasets when overdispersion parameter is bounded and I estimate the sequencing error rate.</a></li>
<li><a href="showoff_plots.html">Show off new methods for updog for plotting and summarizing.</a></li>
<li><a href="vanilla_updog.html">First attempt at modeling bias parameter</a>. Looks promising, but still need to do a lot of work.</li>
<li><a href="prob_lines.html">Plot probability lines at different levels of seq_error and bias_val</a>.</li>
<li><a href="vanilla_updog_outlier.html">First fits with outlier model.</a> Things look pretty good!</li>
<li><a href="diagnostic_plot.html">Some EDA from bad SNP</a> Re-ran after I made it so <code>out_mean</code> is constrained to be 0.5. It looks pretty good now!</li>
<li><a href="vanilla_updog_out_mean_06.html">Updog where mean of outlier distribution is constrained to be 0.5</a></li>
<li><a href="presentation_figures.html">Presentation figures.</a> There are a lot of good inference results here.</li>
<li><a href="check_parental_precisions.html">Messy first look at parental data</a>.</li>
<li><a href="parent_info.html">Diagnostics on the super overdispersed SNP</a>. It seems the best measure to remove this SNP is <code>maxpostprob</code>. This is very lower (max at 0.73 and average a little less than that).</li>
<li><a href="blischak_compare.html">First try at Blischak software.</a> I had a mistake in this code. Ignore these results.</li>
<li><a href="blischak_compare_fixed.html">Second try at Blischak software.</a> I prefer the visual display of updog better.</li>
<li><a href="shirasawa_eda.html">EDA of Shirasawa data</a> I think I have the SNPs I want to demonstrate the method on.</li>
<li><a href="shirasawa_est_p1.html">Same as above to estimate the parental genotype.</a> Each time I get 5 as the parental genotype.</li>
<li><a href="updog_sum_stats.html">Histograms of updog fits in Shirasawa data.</a> OD isn’t too large.</li>
<li><a href="updog_ash.html">Fit ASH to updog estimates</a> Don’t know how worthwhile it is.</li>
<li><a href="oracle_screen.html">Screen Via Oracle Estimator</a></li>
<li><a href="check_od_thresh.html">OD estimate weakly correlates with genotyping performance when OD is constant and not too large</a>.</li>
<li><a href="autotetraploid_init.html">Autotetraploid species</a> Looks messy.</li>
<li><a href="tet_explore.html">More autotetraploid species</a>. This looks cleaner. There seems to be less bias in these data, but they are still messy. EBG and UPDOG appear to perform about the same. I used uniform updog while only updating the sequencing error rate. I did this because other genotype models don’t make sense for these data and I was woried about the strong genotype distribution assumptions when estimating read mapping bias, overdispersion, and outlying observations. I think I’ll need the pedigree info to get a better idea on how well these methods are performing.</li>
</ul>
</div>
<div id="alfalfa-project" class="section level1">
<h1>Alfalfa Project:</h1>
<ul>
<li><a href="parse_vcf.html">Parse vcf files, extract readcounts, filter out all variants with more than two alleles</a>. This file contains some EDA on the VCF files. But mostly, this code gets the read-count matrices.</li>
<li><a href="initial_updog_fits.html">Initial fits for some SNPs</a>. These data seem messier than the sweet potato data. Though, there seems to be much less sequencing error. Bias is still a possible problem for some SNPs.</li>
<li><a href="tons_of_plots.html">Tons of raw plots</a> Tons of raw plots. I am trying to see features of the data.</li>
<li><a href="initial_updog_fits_with_bias.html">Try the bias corrected updog on the alfalfa data.</a> Some things look good, some things look bad.</li>
<li><a href="seq_error_prior.html">Motivate Prior on sequencing error</a>. I think a reasonable prior is Beta(mean = 0.005, od = 0.02). This is motivated by the sequencing error rates that the Blischak paper assumes.</li>
<li><a href="seq_error_prior_nonbeta.html">Better prior on sequencing errorr rate</a>. Here, I just use a logit-normal. This is better because the beta has an infinite spike at 0 for reasonable priors. The logit normal works much nicer in practice and is intuitive.</li>
<li><a href="bias_prior.html">Motivate Prior on bias parameter</a>. I think a reasonable prior is N(0, 1) on the log-bias parameter. Values of -3 and 3 are super extreme and I would consider to be extremely poorly behaved SNPs that I would probably throw out if we ever saw them.</li>
</ul>
</div>
<div id="non-sibling-project" class="section level1">
<h1>Non-sibling Project</h1>
<ul>
<li><a href="blischak_data_eda.html">Data that Blischak uses has super low coverage!</a></li>
<li><a href="uitdewilligen_eda.html">EDA on Uitdewillegen Data.</a> It’s not as clean as the sweet potato.</li>
<li><a href="local_minima.html">Updog can run into local minima if the bias is high and the od is low.</a></li>
</ul>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
