<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="David Gerard" />

<meta name="date" content="2017-04-25" />

<title>Initial Updog Fits</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">temp</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Initial Updog Fits</h1>
<h4 class="author"><em>David Gerard</em></h4>
<h4 class="date"><em>April 25, 2017</em></h4>

</div>


<div id="abstract" class="section level1">
<h1>Abstract</h1>
<p>Here, I am going to run some EDA on a few alfalfa SNP’s. I want to see if I observe the same problems that I saw in sweet potatoes. Namely, do I observe:</p>
<ol style="list-style-type: decimal">
<li>Sequencing errors,</li>
<li>Overdispersion,</li>
<li>Outliers</li>
<li>Bias</li>
</ol>
<p>Hopefully, I won’t see systematic Bias. Since there is no reference genome, we won’t be able to use WASP, so I would probably need to model this bias directly.</p>
</div>
<div id="subset-data" class="section level1">
<h1>Subset Data</h1>
<pre class="r"><code>obj &lt;- readRDS(file = &quot;../output/allele_counts/chr1_ct.RDS&quot;)
str(obj)</code></pre>
<pre><code>## List of 3
##  $ fixMat:Classes &#39;tbl_df&#39;, &#39;tbl&#39; and &#39;data.frame&#39;:  62076 obs. of  7 variables:
##   ..$ CHROM : chr [1:62076] &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; ...
##   ..$ POS   : chr [1:62076] &quot;6617&quot; &quot;6634&quot; &quot;6640&quot; &quot;6641&quot; ...
##   ..$ ID    : chr [1:62076] &quot;S1_6617&quot; &quot;S1_6634&quot; &quot;S1_6640&quot; &quot;S1_6641&quot; ...
##   ..$ REF   : chr [1:62076] &quot;G&quot; &quot;G&quot; &quot;T&quot; &quot;A&quot; ...
##   ..$ ALT   : chr [1:62076] &quot;A&quot; &quot;A&quot; &quot;-&quot; &quot;-&quot; ...
##   ..$ QUAL  : chr [1:62076] NA NA NA NA ...
##   ..$ FILTER: chr [1:62076] &quot;PASS&quot; &quot;PASS&quot; &quot;PASS&quot; &quot;PASS&quot; ...
##  $ A     : num [1:62076, 1:412] 5 5 5 5 5 5 5 5 5 5 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:62076] &quot;S1_6617&quot; &quot;S1_6634&quot; &quot;S1_6640&quot; &quot;S1_6641&quot; ...
##   .. ..$ : chr [1:412] &quot;DM3.1:SRR1266111:7:124151&quot; &quot;DM3.10:SRR1586329:6:123939&quot; &quot;DM3.11:SRR1586330:1:124045&quot; &quot;DM3.12:SRR1586330:1:124046&quot; ...
##  $ a     : num [1:62076, 1:412] 0 0 0 0 0 0 0 0 0 0 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:62076] &quot;S1_6617&quot; &quot;S1_6634&quot; &quot;S1_6640&quot; &quot;S1_6641&quot; ...
##   .. ..$ : chr [1:412] &quot;DM3.1:SRR1266111:7:124151&quot; &quot;DM3.10:SRR1586329:6:123939&quot; &quot;DM3.11:SRR1586330:1:124045&quot; &quot;DM3.12:SRR1586330:1:124046&quot; ...</code></pre>
<p>I will choose SNP’s that are at various levels of total read-counts.</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
med_counts &lt;- apply(obj$A + obj$a, 1, median, na.rm = TRUE)
mean_counts &lt;- rowMeans(obj$A + obj$a, na.rm = TRUE)
qplot(log10(med_counts), geom = &quot;histogram&quot;, color = I(&quot;black&quot;), 
      fill = I(&quot;white&quot;), bins = 30, xlab = &quot;Log10-median-read-counts&quot;, ylab = &quot;Counts&quot;) + theme_bw()</code></pre>
<pre><code>## Warning: Removed 25 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>qplot(log10(mean_counts), geom = &quot;histogram&quot;, color = I(&quot;black&quot;), 
      fill = I(&quot;white&quot;), bins = 30, xlab = &quot;Log10-median-read-counts&quot;, ylab = &quot;Counts&quot;) + theme_bw()</code></pre>
<pre><code>## Warning: Removed 25 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<p>There seem to be 25 variants with all missing data.</p>
<pre class="r"><code>sum(is.na(med_counts))</code></pre>
<pre><code>## [1] 25</code></pre>
<pre class="r"><code>quants &lt;- quantile(mean_counts, probs = seq(0, 1, length = 30), na.rm = TRUE)
quants</code></pre>
<pre><code>##      0%  3.448%  6.897%  10.34%  13.79%  17.24%  20.69%  24.14%  27.59% 
##   1.000   1.239   1.311   1.381   1.463   1.538   1.618   1.700   1.782 
##  31.03%  34.48%  37.93%  41.38%  44.83%  48.28%  51.72%  55.17%  58.62% 
##   1.884   1.992   2.113   2.283   2.482   2.751   3.035   3.392   3.848 
##  62.07%  65.52%  68.97%  72.41%  75.86%  79.31%  82.76%  86.21%  89.66% 
##   4.377   5.053   5.956   7.052   8.340   9.764  11.686  14.154  17.864 
##   93.1%  96.55%    100% 
##  22.253  31.789 285.833</code></pre>
<pre class="r"><code>match_vec &lt;- match(round(quants, digits = 1), round(mean_counts, digits = 1))
mean_counts[match_vec]</code></pre>
<pre><code>##  S1_1625431   S1_102599   S1_102235   S1_105875   S1_108412   S1_108412 
##       1.046       1.193       1.253       1.409       1.487       1.487 
##    S1_44443   S1_222037    S1_75323    S1_75297   S1_227405   S1_114834 
##       1.606       1.739       1.750       1.871       2.030       2.143 
##   S1_147671   S1_103373   S1_202789   S1_114490  S1_1511556   S1_107815 
##       2.277       2.452       2.828       3.028       3.422       3.779 
##   S1_139511   S1_189421   S1_814259    S1_19481   S1_217570  S1_2426238 
##       4.391       5.084       5.975       7.052       8.308       9.751 
##  S1_1660253  S1_4112241  S1_4320691  S1_1866552  S1_6322702 S1_24061315 
##      11.666      14.175      17.876      22.255      31.772     285.833</code></pre>
<pre class="r"><code>sub_A &lt;- obj$A[match_vec, ]
sub_a &lt;- obj$a[match_vec, ]
rowMeans(sub_a + sub_A, na.rm = TRUE)</code></pre>
<pre><code>##  S1_1625431   S1_102599   S1_102235   S1_105875   S1_108412   S1_108412 
##       1.046       1.193       1.253       1.409       1.487       1.487 
##    S1_44443   S1_222037    S1_75323    S1_75297   S1_227405   S1_114834 
##       1.606       1.739       1.750       1.871       2.030       2.143 
##   S1_147671   S1_103373   S1_202789   S1_114490  S1_1511556   S1_107815 
##       2.277       2.452       2.828       3.028       3.422       3.779 
##   S1_139511   S1_189421   S1_814259    S1_19481   S1_217570  S1_2426238 
##       4.391       5.084       5.975       7.052       8.308       9.751 
##  S1_1660253  S1_4112241  S1_4320691  S1_1866552  S1_6322702 S1_24061315 
##      11.666      14.175      17.876      22.255      31.772     285.833</code></pre>
<p>For the smaller counts alleles, most of the samples don’t have any read-counts (indicated by <code>NA</code>) (though, this isn’t a guarantee).</p>
<pre class="r"><code>plot(rowMeans(is.na(sub_a)), type = &quot;l&quot;)</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>plot(rowMeans(is.na(sub_A)), type = &quot;l&quot;)</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
</div>
<div id="raw-plots" class="section level1">
<h1>Raw Plots</h1>
<p>First, some plots of the maximum expressed variants</p>
<pre class="r"><code>library(updog)
ploidy  &lt;- 4
ocounts &lt;- sub_A
osize   &lt;- sub_A + sub_a

for (index in 1:nrow(ocounts)) {
 pl &lt;- plot_geno(ocounts = ocounts[index, ], osize = osize[index, ],
                 ploidy = 4, seq_error = 0)  
 print(pl)
}</code></pre>
<pre><code>## Warning: Removed 347 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre><code>## Warning: Removed 298 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-2.png" width="672" /></p>
<pre><code>## Warning: Removed 329 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-3.png" width="672" /></p>
<pre><code>## Warning: Removed 275 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-4.png" width="672" /></p>
<pre><code>## Warning: Removed 299 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-5.png" width="672" /></p>
<pre><code>## Warning: Removed 299 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-6.png" width="672" /></p>
<pre><code>## Warning: Removed 341 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-7.png" width="672" /></p>
<pre><code>## Warning: Removed 209 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-8.png" width="672" /></p>
<pre><code>## Warning: Removed 264 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-9.png" width="672" /></p>
<pre><code>## Warning: Removed 202 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-10.png" width="672" /></p>
<pre><code>## Warning: Removed 114 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-11.png" width="672" /></p>
<pre><code>## Warning: Removed 321 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-12.png" width="672" /></p>
<pre><code>## Warning: Removed 217 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-13.png" width="672" /></p>
<pre><code>## Warning: Removed 226 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-14.png" width="672" /></p>
<pre><code>## Warning: Removed 68 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-15.png" width="672" /></p>
<pre><code>## Warning: Removed 59 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-16.png" width="672" /></p>
<pre><code>## Warning: Removed 47 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-17.png" width="672" /></p>
<pre><code>## Warning: Removed 132 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-18.png" width="672" /></p>
<pre><code>## Warning: Removed 62 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-19.png" width="672" /></p>
<pre><code>## Warning: Removed 17 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-20.png" width="672" /></p>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-21.png" width="672" /></p>
<pre><code>## Warning: Removed 49 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-22.png" width="672" /></p>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-23.png" width="672" /></p>
<pre><code>## Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-24.png" width="672" /></p>
<pre><code>## Warning: Removed 32 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-25.png" width="672" /><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-26.png" width="672" /><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-27.png" width="672" /><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-28.png" width="672" /><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-29.png" width="672" /><img src="initial_updog_fits_files/figure-html/unnamed-chunk-6-30.png" width="672" /></p>
<p>This last SNP makes no sense. Why are they all clustered at 0.75? Is this a technical artifact? This would only make sense biologically if segregation happened in pairs. E.g. if the parental genotypes were AAAA and ABAB and AB have to cluster together in the second parent. If they all segregated independently, then you would expect to see some AABB’s and some AAAA’s. This is weird.</p>
<p>Of course, when we fit <code>updog</code>, we get weird results.</p>
<pre class="r"><code>uout &lt;- updog(ocounts = ocounts[nrow(ocounts), ], osize = osize[nrow(osize), ],
              seq_error = 0.01, ploidy = 4, update_geno = TRUE, update_outlier = TRUE,
              update_pi = TRUE, update_rho = TRUE)
plot(uout)</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre><code>## Press [enter] to continue</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-7-2.png" width="672" /></p>
<pre><code>## Press [enter] to continue</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-7-3.png" width="672" /></p>
<p>Some of these don’t look too bad!</p>
<pre class="r"><code>for (index in 20:29) {
  ocounts_nona &lt;- ocounts[index, ]
  ocounts_nona &lt;- ocounts_nona[!is.na(ocounts_nona)]
  osize_nona &lt;- osize[index, ]
  osize_nona &lt;- osize_nona[!is.na(osize_nona)]

  saveRDS(file = &quot;alpha_err_dat.RDS&quot;, object = list(ocounts = ocounts_nona, osize = osize_nona, ploidy = 4))

  suppressMessages(uout &lt;- updog(ocounts = ocounts_nona, osize = osize_nona, ploidy = ploidy,
                                 overdispersion = FALSE, update_geno = TRUE))
  pl &lt;- plot(uout, plot_beta = FALSE)
  print(pl)
  pl &lt;- plot_geno(ocounts = ocounts_nona, osize = osize_nona, ploidy = 4, seq_error = 0)
  print(pl)
}</code></pre>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-2.png" width="672" /></p>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-3.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-4.png" width="672" /></p>
<pre><code>## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3 
## 3 3</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-5.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-6.png" width="672" /></p>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-7.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-8.png" width="672" /></p>
<pre><code>## 1 4 
## 1 4 
## 1 4 
## 1 4 
## 1 4 
## 1 4 
## 1 4 
## 1 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-9.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-10.png" width="672" /></p>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-11.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-12.png" width="672" /></p>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-13.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-14.png" width="672" /></p>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-15.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-16.png" width="672" /></p>
<pre><code>## 2 3 
## 2 3 
## 2 3 
## 2 3</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-17.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-18.png" width="672" /></p>
<pre><code>## 3 4 
## 3 4 
## 3 4 
## 3 4 
## 3 4</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-19.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><img src="initial_updog_fits_files/figure-html/unnamed-chunk-8-20.png" width="672" /></p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.2 (2016-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.2 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] updog_0.1.0     dplyr_0.5.0     purrr_0.2.2     readr_1.0.0    
## [5] tidyr_0.6.1     tibble_1.2      ggplot2_2.2.1   tidyverse_1.1.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.10     plyr_1.8.4       forcats_0.2.0    tools_3.3.2     
##  [5] digest_0.6.12    jsonlite_1.3     lubridate_1.6.0  evaluate_0.10   
##  [9] nlme_3.1-131     gtable_0.2.0     lattice_0.20-34  psych_1.6.12    
## [13] DBI_0.6          yaml_2.1.14      parallel_3.3.2   haven_1.0.0     
## [17] xml2_1.1.1       stringr_1.2.0    httr_1.2.1       knitr_1.15.1    
## [21] hms_0.3          rprojroot_1.2    grid_3.3.2       R6_2.2.0        
## [25] readxl_0.1.1     foreign_0.8-67   rmarkdown_1.3    modelr_0.1.0    
## [29] reshape2_1.4.2   magrittr_1.5     codetools_0.2-15 backports_1.0.5 
## [33] scales_0.4.1     htmltools_0.3.5  rvest_0.3.2      assertthat_0.2.0
## [37] mnormt_1.5-5     colorspace_1.3-2 labeling_0.3     stringi_1.1.2   
## [41] lazyeval_0.2.0   munsell_0.4.3    broom_0.4.2</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
