<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="David Gerard" />

<meta name="date" content="2017-04-25" />

<title>Initial Updog Fits</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">temp</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Initial Updog Fits</h1>
<h4 class="author"><em>David Gerard</em></h4>
<h4 class="date"><em>April 25, 2017</em></h4>

</div>


<div id="abstract" class="section level1">
<h1>Abstract</h1>
<p>I run the bias-modeling updog on the alfalfa data.</p>
</div>
<div id="subset-data" class="section level1">
<h1>Subset Data</h1>
<pre class="r"><code>obj &lt;- readRDS(file = &quot;../output/allele_counts/chr1_ct.RDS&quot;)
str(obj)</code></pre>
<pre><code>## List of 2
##  $ A: num [1:62076, 1:412] 5 5 5 5 5 5 5 5 5 5 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:62076] &quot;S1_6617&quot; &quot;S1_6634&quot; &quot;S1_6640&quot; &quot;S1_6641&quot; ...
##   .. ..$ : chr [1:412] &quot;DM3.1:SRR1266111:7:124151&quot; &quot;DM3.10:SRR1586329:6:123939&quot; &quot;DM3.11:SRR1586330:1:124045&quot; &quot;DM3.12:SRR1586330:1:124046&quot; ...
##  $ a: num [1:62076, 1:412] 0 0 0 0 0 0 0 0 0 0 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:62076] &quot;S1_6617&quot; &quot;S1_6634&quot; &quot;S1_6640&quot; &quot;S1_6641&quot; ...
##   .. ..$ : chr [1:412] &quot;DM3.1:SRR1266111:7:124151&quot; &quot;DM3.10:SRR1586329:6:123939&quot; &quot;DM3.11:SRR1586330:1:124045&quot; &quot;DM3.12:SRR1586330:1:124046&quot; ...</code></pre>
<p>I will choose SNP’s that are at various levels of total read-counts.</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
med_counts &lt;- apply(obj$A + obj$a, 1, median, na.rm = TRUE)
mean_counts &lt;- rowMeans(obj$A + obj$a, na.rm = TRUE)
quants &lt;- quantile(mean_counts, probs = seq(0, 1, length = 30), na.rm = TRUE)
match_vec &lt;- match(round(quants, digits = 1), round(mean_counts, digits = 1))
sub_A &lt;- obj$A[match_vec, ]
sub_a &lt;- obj$a[match_vec, ]</code></pre>
<p>For the smaller counts alleles, most of the samples don’t have any read-counts (indicated by <code>NA</code>) (though, this isn’t a guarantee).</p>
<pre class="r"><code>plot(rowMeans(is.na(sub_a)), type = &quot;l&quot;)</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>plot(rowMeans(is.na(sub_A)), type = &quot;l&quot;)</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>
</div>
<div id="raw-plots" class="section level1">
<h1>Raw Plots</h1>
<p>First, some plots of the maximum expressed variants</p>
<pre class="r"><code>library(updog)
ploidy  &lt;- 4
ocounts &lt;- sub_A
osize   &lt;- sub_A + sub_a

for (index in 1:nrow(ocounts)) {
 pl &lt;- plot_geno(ocounts = ocounts[index, ], osize = osize[index, ],
                 ploidy = 4, seq_error = 0)  
 print(pl)
}</code></pre>
<pre><code>## Warning: Removed 347 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre><code>## Warning: Removed 298 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
<pre><code>## Warning: Removed 329 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-3.png" width="672" /></p>
<pre><code>## Warning: Removed 275 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-4.png" width="672" /></p>
<pre><code>## Warning: Removed 299 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-5.png" width="672" /></p>
<pre><code>## Warning: Removed 299 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-6.png" width="672" /></p>
<pre><code>## Warning: Removed 341 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-7.png" width="672" /></p>
<pre><code>## Warning: Removed 209 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-8.png" width="672" /></p>
<pre><code>## Warning: Removed 264 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-9.png" width="672" /></p>
<pre><code>## Warning: Removed 202 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-10.png" width="672" /></p>
<pre><code>## Warning: Removed 114 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-11.png" width="672" /></p>
<pre><code>## Warning: Removed 321 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-12.png" width="672" /></p>
<pre><code>## Warning: Removed 217 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-13.png" width="672" /></p>
<pre><code>## Warning: Removed 226 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-14.png" width="672" /></p>
<pre><code>## Warning: Removed 68 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-15.png" width="672" /></p>
<pre><code>## Warning: Removed 59 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-16.png" width="672" /></p>
<pre><code>## Warning: Removed 47 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-17.png" width="672" /></p>
<pre><code>## Warning: Removed 132 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-18.png" width="672" /></p>
<pre><code>## Warning: Removed 62 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-19.png" width="672" /></p>
<pre><code>## Warning: Removed 17 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-20.png" width="672" /></p>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-21.png" width="672" /></p>
<pre><code>## Warning: Removed 49 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-22.png" width="672" /></p>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-23.png" width="672" /></p>
<pre><code>## Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-24.png" width="672" /></p>
<pre><code>## Warning: Removed 32 rows containing missing values (geom_point).</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-25.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-26.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-27.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-28.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-29.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-4-30.png" width="672" /></p>
<p>This last SNP makes no sense. Why are they all clustered at 0.75? Is this a technical artifact? This would only make sense biologically if segregation happened in pairs. E.g. if the parental genotypes were AAAA and ABAB and AB have to cluster together in the second parent. If they all segregated independently, then you would expect to see some AABB’s and some AAAA’s. This is weird.</p>
<p>An alternative explanation is an extreme sequencing error rate combined with an extreme mapping bias toward the non-reference allele. This is what <code>updog_vanilla</code> catches.</p>
<pre class="r"><code>uout &lt;- updog_vanilla(ocounts = ocounts[nrow(ocounts), ], osize = osize[nrow(osize), ],
                      ploidy = 4)
plot(uout, ask = FALSE)</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-5-1.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-5-2.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-5-3.png" width="672" /></p>
<p>Some of these don’t look too bad!</p>
<pre class="r"><code>for (index in 20:29) {
  ocounts_nona &lt;- ocounts[index, ]
  ocounts_nona &lt;- ocounts_nona[!is.na(ocounts_nona)]
  osize_nona &lt;- osize[index, ]
  osize_nona &lt;- osize_nona[!is.na(osize_nona)]

  uout &lt;- updog_vanilla(ocounts = ocounts_nona, osize = osize_nona, ploidy = ploidy, print_val = FALSE,
                        update_outdisp = FALSE, update_outmean = FALSE, update_outprop = TRUE,
                        non_mono_max = 1)
  plot(uout, ask = FALSE, plot_beta = FALSE)
  print(plot_geno(ocounts = ocounts_nona, osize = osize_nona, ploidy = 4, seq_error = 0))
}</code></pre>
<p><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-1.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-2.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-3.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-4.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-5.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-6.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-7.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-8.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-9.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-10.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-11.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-12.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-13.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-14.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-15.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-16.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-17.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-18.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-19.png" width="672" /><img src="initial_updog_fits_with_bias_files/figure-html/unnamed-chunk-6-20.png" width="672" /></p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.3.2 (2016-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.2 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] updog_0.1.0     dplyr_0.5.0     purrr_0.2.2     readr_1.0.0    
## [5] tidyr_0.6.1     tibble_1.2      ggplot2_2.2.1   tidyverse_1.1.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.10     plyr_1.8.4       forcats_0.2.0    tools_3.3.2     
##  [5] digest_0.6.12    jsonlite_1.3     lubridate_1.6.0  evaluate_0.10   
##  [9] nlme_3.1-131     gtable_0.2.0     lattice_0.20-34  psych_1.6.12    
## [13] DBI_0.6          yaml_2.1.14      parallel_3.3.2   haven_1.0.0     
## [17] xml2_1.1.1       stringr_1.2.0    httr_1.2.1       knitr_1.15.1    
## [21] hms_0.3          rprojroot_1.2    grid_3.3.2       R6_2.2.0        
## [25] readxl_0.1.1     foreign_0.8-67   rmarkdown_1.3    modelr_0.1.0    
## [29] reshape2_1.4.2   magrittr_1.5     codetools_0.2-15 backports_1.0.5 
## [33] scales_0.4.1     htmltools_0.3.5  rvest_0.3.2      assertthat_0.2.0
## [37] mnormt_1.5-5     colorspace_1.3-2 labeling_0.3     stringi_1.1.2   
## [41] lazyeval_0.2.0   munsell_0.4.3    broom_0.4.2</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
