---
title: "Home"
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`


# Sweet Potato Project

Much of this code doesn't work anymore because of modifications to the [`updog`](https://github.com/dcgerard/updog) package

* [First analysis of SNP data](snp_analysis.html)
* [Look at iterative procedure and effect of uniform prior on parents](iter_procedure.html). This results in a non-uniform prior on the children, as can be expected.
* [First try at overdispersed updog](updot_overdispersion.html)
* [Run overdispersed updog on all of the datasets Felipe gave me](more_updog_over.html)
* [Look at estimating sequencing error rate and results look promising. No more errors when I bound the overdispersion parameter.](updog_overdispersion_seq_est.html)
* [Run through all datasets when overdispersion parameter is bounded and I estimate the sequencing error rate.](more_updog_over_seq_est.html)
* [Show off new methods for updog for plotting and summarizing.](showoff_plots.html)
* [First attempt at modeling bias parameter](vanilla_updog.html). Looks promising, but still need to do a lot of work.
* [Plot probability lines at different levels of seq_error and bias_val](prob_lines.html).
* [First fits with outlier model.](vanilla_updog_outlier.html) Things look pretty good!
* [Some EDA from bad SNP](diagnostic_plot.html) Re-ran after I made it so `out_mean` is constrained to be 0.5. It looks pretty good now!
* [Updog where mean of outlier distribution is constrained to be 0.5](vanilla_updog_out_mean_06.html)
* [Presentation figures.](presentation_figures.html) There are a lot of good inference results here.
* [Messy first look at parental data](check_parental_precisions.html).
* [Diagnostics on the super overdispersed SNP](parent_info.html). It seems the best measure to remove this SNP is `maxpostprob`. This is very lower (max at 0.73 and average a little less than that).
* [First try at Blischak software.](blischak_compare.html) I had a mistake in this code. Ignore these results.
* [Second try at Blischak software.](blischak_compare_fixed.html) I prefer the visual display of updog better.
* [EDA of Shirasawa data](shirasawa_eda.html) I think I have the SNPs I want to demonstrate the method on.
* [Same as above to estimate the parental genotype.](shirasawa_est_p1.html) Each time I get 5 as the parental genotype.
* [Histograms of updog fits in Shirasawa data.](updog_sum_stats.html) OD isn't too large.
* [Fit ASH to updog estimates](updog_ash.html) Don't know how worthwhile it is.
* [Screen Via Oracle Estimator](oracle_screen.html)
* [OD estimate weakly correlates with genotyping performance when OD is constant and not too large](check_od_thresh.html).
* [Autotetraploid species](autotetraploid_init.html) Looks messy.
* [More autotetraploid species](tet_explore.html). This looks cleaner. There seems to be less bias in these data, but they are still messy. EBG and UPDOG appear to perform about the same. I used uniform updog while only updating the sequencing error rate. I did this because other genotype models don't make sense for these data and I was woried about the strong genotype distribution assumptions when estimating read mapping bias, overdispersion, and outlying observations. I think I'll need the pedigree info to get a better idea on how well these methods are performing.

# Alfalfa Project:

* [Parse vcf files, extract readcounts, filter out all variants with more than two alleles](parse_vcf.html). This file contains some EDA on the VCF files. But mostly, this code gets the read-count matrices.
* [Initial fits for some SNPs](initial_updog_fits.html). These data seem messier than the sweet potato data. Though, there seems to be much less sequencing error. Bias is still a possible problem for some SNPs.
* [Tons of raw plots](tons_of_plots.html) Tons of raw plots. I am trying to see features of the data.
* [Try the bias corrected updog on the alfalfa data.](initial_updog_fits_with_bias.html) Some things look good, some things look bad.
* [Motivate Prior on sequencing error](seq_error_prior.html). I think a reasonable prior is Beta(mean = 0.005, od = 0.02). This is motivated by the sequencing error rates that the Blischak paper assumes.
* [Better prior on sequencing errorr rate](seq_error_prior_nonbeta.html). Here, I just use a logit-normal. This is better because the beta has an infinite spike at 0 for reasonable priors. The logit normal works much nicer in practice and is intuitive.
* [Motivate Prior on bias parameter](bias_prior.html). I think a reasonable prior is N(0, 1) on the log-bias parameter. Values of -3 and 3 are super extreme and I would consider to be extremely poorly behaved SNPs that I would probably throw out if we ever saw them.

# Non-sibling Project
* [Data that Blischak uses has super low coverage!](blischak_data_eda.html)
* [EDA on Uitdewillegen Data.](uitdewilligen_eda.html) It's not as clean as the sweet potato.
* [Updog can run into local minima if the bias is high and the od is low.](local_minima.html)
