<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="David Gerard" />

<meta name="date" content="2016-11-28" />

<title>Look at updog performance when estimating sequencing error</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script src="libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">updog_sims</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="LICENSE">License</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Look at updog performance when estimating sequencing error</h1>
<h4 class="author"><em>David Gerard</em></h4>
<h4 class="date"><em>2016-11-28</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#read-in-snps">Read in SNPs</a></li>
<li><a href="#description-of-updates">Description of updates</a></li>
<li><a href="#description-of-estimating-the-sequencing-error-rate.">Description of estimating the sequencing error rate.</a></li>
<li><a href="#re-run-updog-on-problem-dataset.">Re-run updog on problem dataset.</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-11-28</p>
<p><strong>Code version:</strong> 012a46db0668ec4ca7c1a37bb42d86ef68c1006b</p>
<p>Here, I look at my new sequencing error estimation procedure and how it plays with updog.</p>
<div id="read-in-snps" class="section level2">
<h2>Read in SNPs</h2>
<pre class="r"><code>library(updog)
set.seed(483)
load(&quot;../data/subset_David.Rdata&quot;)
dat &lt;- subset_david
lastp &lt;- 12 ## number of final rows that are parents

## Run `updog`
cmat &lt;- dat[[7]]
p1dat &lt;- cmat[(nrow(cmat) - lastp + 1):(nrow(cmat) - lastp / 2), ]
p2dat &lt;- cmat[(nrow(cmat) - lastp / 2 + 1):nrow(cmat), ]
odat  &lt;- cmat[1:(nrow(cmat) - lastp), ]

p1counts &lt;- p1dat[, 1]
p1size   &lt;- rowSums(p1dat)
p2counts &lt;- p2dat[, 1]
p2size   &lt;- rowSums(p2dat)
ocounts  &lt;- odat[, 1]
osize    &lt;- rowSums(odat)
ploidy &lt;- 6</code></pre>
</div>
<div id="description-of-updates" class="section level2">
<h2>Description of updates</h2>
I think I was having trouble because of something weird with the beta-binomial. For some values of the shape parameters (when either is less than 1), you can get the highest densities on the extremes 0 and 1. This is at the very least non-intuitive behavior for an overdispersed model. To counter this, I bound the overdispersion parameter so that the beta-density is finite at 0 and 1 for the smallest value of the non-sequencing error probabilities. That is, I find <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> such that
<span class="math display">\[\begin{align}
\frac{1}{ploidy} &amp;= \frac{\alpha}{\alpha + \beta},\\
\beta &amp;\geq \alpha \geq 1.
\end{align}\]</span>
The solution to this is to set <span class="math inline">\(\alpha = 1\)</span> and <span class="math inline">\(\beta = ploidy - 1\)</span>. In terms of the mean (<span class="math inline">\(\mu\)</span>) and overdispersion parameter (<span class="math inline">\(\rho\)</span>) of the beta-binomial, this is equivalent to setting
<span class="math display">\[\begin{align}
\mu &amp;= \frac{1}{ploidy} \\
\rho &amp;= \frac{1}{ploidy + 1}.
\end{align}\]</span>
<p>Hence, I bound the overdispersion parameter to be a maximum of 1 / (ploidy + 1).</p>
</div>
<div id="description-of-estimating-the-sequencing-error-rate." class="section level2">
<h2>Description of estimating the sequencing error rate.</h2>
<p>I think we might be able to estimate this by maximum likelihood, but I also have a heuristic scheme to do so too. Let <span class="math inline">\(\hat{p}\)</span> be the estimated proportions. Keep the values such that <span class="math inline">\(\hat{p} \geq 1 - \epsilon\)</span> for some fixed <span class="math inline">\(\epsilon\)</span>, then estimate the probability of an “a” given these values. The idea is that the reference allele will be common enough to have many offspring with all “A”’s, in which case we can look how often these observations deviate from all A’s.</p>
</div>
<div id="re-run-updog-on-problem-dataset." class="section level2">
<h2>Re-run updog on problem dataset.</h2>
<pre class="r"><code>uout1 &lt;- updog(ocounts = ocounts, osize = osize, ploidy = ploidy,
              update_rho = TRUE, overdispersion = TRUE,
              update_geno = TRUE, update_outlier = TRUE, update_pi = TRUE)</code></pre>
<p>The estimated sequencing error was 0.0047.</p>
<pre class="r"><code>pl &lt;- plot_geno(ocounts = ocounts, osize = osize, ploidy = ploidy, p1counts = p1counts, p1size = p1size,
                p2counts = p2counts, p2size = p2size, col = uout1$ogeno, prob_ok = uout1$prob_ok)
print(pl, seq_error = uout1$seq_error)</code></pre>
<p><img src="figure/updog_overdispersion_seq_est.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There are no errors here and I am getting something that looks a lot more intuitive. The estimated overdispersion parameter is 0.0046.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] updog_0.1.0 knitr_1.15 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.7      digest_0.6.10    assertthat_0.1   plyr_1.8.4      
 [5] grid_3.3.2       gtable_0.2.0     magrittr_1.5     evaluate_0.10   
 [9] scales_0.4.1     ggplot2_2.2.0    stringi_1.1.2    lazyeval_0.2.0  
[13] rmarkdown_1.1    labeling_0.3     tools_3.3.2      stringr_1.1.0   
[17] munsell_0.4.3    yaml_2.1.14      colorspace_1.3-0 htmltools_0.3.5 
[21] tibble_1.2      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
